server/src/resources/itinerary_flight_plan/mod.rs

```diff
@@ -0,0 +1,104 @@
+//! Itineraries
+
+/// module providing functions to generate mock data
+pub mod mock;
+
+mod grpc_server {
+    #![allow(unused_qualifications)]
+    tonic::include_proto!("grpc.itinerary_flight_plan");
+}
+// Expose module resources
+pub use grpc_server::rpc_service_server::*;
+pub use grpc_server::*;
+
+use core::fmt::Debug;
+use log::debug;
+use std::collections::HashMap;
+use tokio_postgres::row::Row;
+use tokio_postgres::types::Type as PsqlFieldType;
+use tonic::{Request, Status};
+use uuid::Uuid;
+
+use crate::common::ArrErr;
+use crate::grpc::{
+    AdvancedSearchFilter, FilterOption, GrpcDataObjectType, GrpcField, GrpcObjectType, Id,
+    PredicateOperator, SearchFilter, ValidationResult,
+};
+use crate::resources::base::{
+    FieldDefinition, GenericObjectType, GenericResource, GenericResourceResult, Resource,
+    ResourceDefinition,
+};
+
+// Generate `From` trait implementations for GenericResource into and from Grpc defined Resource
+crate::build_generic_resource_impl_from!();
+
+// Generate grpc server implementations
+crate::build_grpc_resource_impl!(itinerary);
+crate::build_grpc_server_generic_impl!(itinerary);
+
+impl TryFrom<Row> for Data {
+    type Error = ArrErr;
+
+    fn try_from(row: Row) -> Result<Self, ArrErr> {
+        debug!("Converting Row to itinerary::Data: {:?}", row);
+        let itinerary_id: String = row.get::<&str, Uuid>("itinerary_id").to_string();
+        let flight_plan_id: String = row.get::<&str, Uuid>("flight_plan_id").to_string();
+
+        Ok(Data {
+            itinerary_id,
+            flight_plan_id,
+        })
+    }
+}
+
+impl Resource for GenericResource<Data> {
+    fn get_definition() -> ResourceDefinition {
+        ResourceDefinition {
+            psql_table: String::from("itinerary_flight_plan"),
+            psql_id_col: String::from("itfp_id"),
```

Comment:
In this case, we'll want to have a combined primary key instead, so we'll need a way to tell our database. We don't want to store duplicate data by introducing a separate column.
I think we might end up with a new struct `LinkedData` or something, with it's own implementations for table creation / updates/ inserts